### PR TITLE
Code improvements for media file downloading

### PR DESCRIPTION
**Pull request by grzesiek2010**
#### Why is this the best possible solution? Were any other approaches considered?
I've introduced sanitizing media file names and removed the support for `InstanceProvider#update` as we discussed with @seadowg.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
The only thing that is worth testing here is downloading forms with media files.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] added or modified tests for any new or changed behavior
- [x] run `./gradlew connectedAndroidTest` (or `./gradlew testLab`) and confirmed all checks still pass
- [x] added a comment above any new strings describing it for translators
- [x] added any new strings with date formatting to `DateFormatsTest`
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

**Review by seadowg — CHANGES_REQUESTED** [review]
This needs to merge into `v2024.2.x` instead of `master`

File: collect_app/src/main/java/org/odk/collect/android/external/InstanceProvider.java [hunk 1]
```diff
@@ -197,67 +193,7 @@ public int delete(@NonNull Uri uri, String where, String[] whereArgs) {
 
     @Override
     public int update(@NonNull Uri uri, ContentValues values, String where, String[] whereArgs) {
-        DaggerUtils.getComponent(getContext()).inject(this);
-
-        String projectId = getProjectId(uri);
-        logServerEvent(projectId, AnalyticsEvents.INSTANCE_PROVIDER_UPDATE);
```
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/external/InstanceProvider.java:203]: This analytics event should be deleted as well.

**Review by seadowg — CHANGES_REQUESTED** [review]

File: collect_app/src/main/java/org/odk/collect/android/external/InstanceProvider.java (hunk 1 above)
Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/external/InstanceProvider.java:203]: Done.

**Review by seadowg — APPROVED** [review]

**Comment by WKobus** [comment]
**Tested with Success!**

Verified on device with Android 14
Dominika verified this PR on Android 10

**Verified Cases:**
- Downloading forms with media files
- Uploading different types of media files
- Updating media files
- Cancelling downloading of forms with media files
- Manual and match exactly update settings

**Pull request merged**